pkg/status/singletonstatus.go

```diff
@@ -18,65 +18,71 @@ package status
 
 import (
 	"context"
-	"fmt"
 
-	"k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 
-	"github.com/kubestellar/kubestellar/api/control/v1alpha1"
 	"github.com/kubestellar/kubestellar/pkg/util"
 )
 
-// reconcileSingletonByBdg goes over the binding-covered workload objects, decides sync or not for each of the workload objects,
-// then maintains their statuses based on the decisions.
 func (c *Controller) reconcileSingletonByBdg(ctx context.Context, bdgName string) error {
 	logger := klog.FromContext(ctx)
 	logger.V(4).Info("Reconciling singleton status due to binding change", "binding", bdgName)
 
-	binding, err := c.wdsKsClient.ControlV1alpha1().Bindings().Get(ctx, bdgName, metav1.GetOptions{})
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// A deleted Binding is equivalent to one that references zero workload objects
-			return nil
+	wObjIDs, requested, nWECs := c.bindingPolicyResolver.GetSingletonReportedStateRequestsForBinding(bdgName)
+	for i := range wObjIDs {
+		wObjID, r, n := wObjIDs[i], requested[i], nWECs[i]
+		if !r || n != 1 {
+			logger.V(4).Info("Workload object should not have singleton status synced, cleaning",
+				"resource", wObjID.Resource, "objectName", wObjID.ObjectName,
```

Comment:
Why `wObjID.Resource` instead of `wObjID.GVK`? We _do_ have cases of one Kind name appearing in multiple API groups. And one day --- if we are lucky enough to get this far --- users will report hard-to-diagnose problems caused by the fact that API group _version_ appears in the keys here.

Reply:
Changed to `wObjID.GVK` everywhere.